### PR TITLE
Beacon block

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

https://github.com/sigp/lighthouse/issues/49

## Proposed Changes

- Rename `Block` -> `BeaconBlock`, globally
- Add new fields to `BeaconBlock` (ancestors, specials, etc)
- Update `SszBeaconBlock` as per new struct layout
- Update all dependant functions (e.g., block_store, validation, etc)

## Additional Info

This is a work-in-progress, please do not review yet.

**Please do not merge before #33 is merged**.

### CODE REVIEW

**Comment by paulhauner** [comment]
Ready for review!

**Please don't merge, just approved. #33 needs to be merged first.**

**Comment by paulhauner** [comment]
Ok, this is ready for review! The diff is much smaller now #33 has been merged.

See PR description for list of changes.

File: beacon_chain/types/src/beacon_block.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+use super::special_record::SpecialRecord;
+use super::ssz::{
+    Encodable,
+    Decodable,
+    DecodeError,
+    SszStream,
+};
+
+pub const MIN_SSZ_BLOCK_LENGTH: usize = {
+    8 +                 // slot
```
Comment (HCastano) [inline beacon_chain/types/src/beacon_block.rs]: Why not do something like `const SLOT: usize = 8;` for all the fields?

**Comment by paulhauner** [comment]
Good call, will do.

Thanks!

File: beacon_chain/types/src/beacon_block.rs (hunk 1 above)
Comment (paulhauner) [reply beacon_chain/types/src/beacon_block.rs]: Implemented. Great point @HCastano

**Pull request merged**